app/lib/actions/trigger_proxy_batch.rb

```diff
@@ -0,0 +1,41 @@
+module Actions
+  class TriggerProxyBatch < Base
+    TriggerNextBatch = Algebrick.atom
+    TriggerLastBatch = Algebrick.atom
+
+    # If the event could result into sub tasks being planned, check if there are any RemoteTasks
+    # to trigger after the event is processed
+    #
+    # The ProxyAction needs to be planned with `:use_batch_triggering => true` to activate the feature
+    def run(event = nil)
+      case event
+      when nil
+        suspend
+      when TriggerNextBatch
+        trigger_remote_tasks_batch and suspend
+      when TriggerLastBatch
+        trigger_remote_tasks_batch
+      end
+    end
+
+    def trigger_remote_tasks_batch
+      # Find the tasks in batches, order them by proxy_url so we get all tasks
+      # to a certain proxy "close to each other"
+      batch = remote_tasks.pending.order(:proxy_url, :id).first(batch_size)
+      # Group the tasks by operation, in theory there should be only one operation
+      batch.group_by(&:operation).each do |operation, group|
+        ForemanTasks::RemoteTask.batch_trigger(operation, group)
```

Comment:
What happens if this raises an exception? I assume this action will go to paused-error. Will anything flip the other action or will it stay suspended till the end of time?

Reply:
I'd guess that all the actions would be planned to wait for getting send over to the proxy, and that would never happen, so all of them would stay waiting forever.
How should I resolve that?

Reply:
In https://github.com/theforeman/foreman-tasks/pull/641/files#diff-0ee6578e875e05026a38f039fe540c5989118323f4f388ca3e4d775ee7e8fe22 I did a similar thing, but turned the middleware into a polling action. How about that?